### ISSUE TITLE
Build failure: torsocks-2.3.0

### ISSUE THREAD

**Issue by sneak** [NixOS/nixpkgs#278619]
### Steps To Reproduce

1. macos arm `nix-env -i tor` (pulls in torsocks as dep)

### Build log

```
make[2]: Entering directory '/private/tmp/nix-build-torsocks-2.3.0.drv-0/torsocks/tests'
  CC       test_dns.o
test_dns.c:95:11: error: call to undeclared function 'gethostbyaddr_r'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        result = gethostbyaddr_r((const void *)&addr,
                 ^
test_dns.c:95:11: note: did you mean 'gethostbyaddr'?
/nix/store/kjcaylsca1fg47cgz94079xpgdfzi8s0-libSystem-11.0.0/include/netdb.h:274:17: note: 'gethostbyaddr' declared here
struct hostent  *gethostbyaddr(const void *, socklen_t, int);
                 ^
test_dns.c:114:11: error: call to undeclared function 'gethostbyaddr_r'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        result = gethostbyaddr_r((const void *)&addr,
                 ^
2 errors generated.
make[2]: *** [Makefile:684: test_dns.o] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-torsocks-2.3.0.drv-0/torsocks/tests'
make[1]: *** [Makefile:713: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-torsocks-2.3.0.drv-0/torsocks/tests'
make: *** [Makefile:468: all-recursive] Error 1
error: builder for '/nix/store/8j4zvxkra1vbqqgq3sqij4y3r6rj8c0g-torsocks-2.3.0.drv' failed with exit code 2;
       last 10 log lines:
       >                  ^
       > test_dns.c:114:11: error: call to undeclared function 'gethostbyaddr_r'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
       >         result = gethostbyaddr_r((const void *)&addr,
       >                  ^
       > 2 errors generated.
       > make[2]: *** [Makefile:684: test_dns.o] Error 1
       > make[2]: Leaving directory '/private/tmp/nix-build-torsocks-2.3.0.drv-0/torsocks/tests'
       > make[1]: *** [Makefile:713: all-recursive] Error 1
       > make[1]: Leaving directory '/private/tmp/nix-build-torsocks-2.3.0.drv-0/torsocks/tests'
       > make: *** [Makefile:468: all-recursive] Error 1
       For full logs, run 'nix-store -l /nix/store/8j4zvxkra1vbqqgq3sqij4y3r6rj8c0g-torsocks-2.3.0.drv'.
error: 1 dependencies of derivation '/nix/store/7zs40nx0fi3sa3z2m0wz3ahlhxbym17h-tor-0.4.8.10.drv' failed to build
```

### Notify maintainers

@vidister 

### Metadata

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

```

---

**Comment by Mayeu** [comment]
I'm also having trouble. I found out there is an upstream ticket for this: https://gitlab.torproject.org/tpo/core/torsocks/-/issues/40005

**Comment by emilytrau** [comment]
Fixed by #287521
